### ISSUE TITLE
ZipArchiveFileLookup.findDirEntry is an allocation hotspot

### ISSUE THREAD

**Issue by retronym** [scala/scala-dev#540]
```scala
  private def findDirEntry(pkg: String): Option[archive.DirEntry] = {
    val dirName = FileUtils.dirPath(pkg) + "/"
    archive.allDirs.get(dirName)
  }
```

 - [ ] On `ZipArchiveFileLookup` creation, index `archive.allDirs` by the dotted package names so that subsequent lookups can be done without temporary strings.

**Comment by retronym** [comment]
WIP: https://github.com/retronym/scala/tree/perf/find-dir-entry

**Comment by mkeskells** [comment]
cant we bypass the experimental io and use the java jar/zip directly for classpath access. There is a small surface API to cover I think

**Comment by diesalbla** [comment]
@retronym It seems that your branch `perf/find-dir-entry` is already merged. Is this still an issue?

**Comment by dwijnand** [comment]
Fixed in scala/scala#7128
